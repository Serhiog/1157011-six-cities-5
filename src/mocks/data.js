
export const cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const offers = [
  {
    id: 0,
    isFavorite: false,
    isPremium: false,
    features: {
      type: `hotel`,
      badrooms: 1,
      adults: 2,
    },
    price: 100,
    title: `TITLE`,
    rating: 3.4,
    things: [
      `Wifi`,
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Own bathroom`,
      `Hair-dryer`,
      `AC`,
    ],
    previewPictures: `img/apartment-small-01.jpg`,
    pictures: [
      `img/apartment-small-03.jpg`,
      `img/apartment-small-04.jpg`,
      `img/room-small.jpg`,
      `img/studio-photos.jpg`,
    ],
    description: `The delusion emphasizes intelligent common sense, although the opposite is accepted in the official language.`,
    owner: {
      id: 0,
      name: `Spider Man`,
      super: true,
      picture: `img/avatar-angelina.jpg`,
    },
    detailsMapZoom: 12,
    coordinates: [52.3909553943508, 4.85309666406198],
    mapZoom: 10,
    city: `Amsterdam`,
    cityCoordinates: [52.38333, 4.9],
  },
  {
    id: 2,
    isFavorite: true,
    isPremium: false,
    features: {
      type: `hotel`,
      badrooms: 2,
      adults: 4,
    },
    price: 200,
    title: `TITLE - 2`,
    rating: 4.5,
    things: [
      `Wifi`,
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Own bathroom`,
      `Hair-dryer`,
      `AC`,
    ],
    previewPictures: `img/apartment-small-01.jpg`,
    pictures: [
      `img/apartment-small-03.jpg`,
      `img/apartment-small-04.jpg`,
      `img/room-small.jpg`,
      `img/studio-photos.jpg`,
    ],
    description: `The delusion emphasizes intelligent common sense, although the opposite is accepted in the official language.`,
    owner: {
      id: 1,
      name: `Iron Man`,
      super: false,
      picture: `img/avatar-angelina.jpg`,
    },
    detailsMapZoom: 12,
    coordinates: [52.3909553943508, 4.8530966640619],
    mapZoom: 10,
    city: `Amsterdam`,
    cityCoordinates: [52.38333, 4.9],
  },
  {
    id: 3,
    isFavorite: false,
    isPremium: true,
    features: {
      type: `private room`,
      badrooms: 1,
      adults: 6,
    },
    price: 300,
    title: `TITLE - 3`,
    rating: 4.9,
    things: [
      `Wifi`,
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Own bathroom`,
      `Hair-dryer`,
      `AC`,
    ],
    previewPictures: `img/apartment-small-01.jpg`,
    pictures: [
      `img/apartment-small-03.jpg`,
      `img/apartment-small-04.jpg`,
      `img/room-small.jpg`,
      `img/studio-photos.jpg`,
    ],
    description: `The delusion emphasizes intelligent common sense, although the opposite is accepted in the official language.`,
    owner: {
      id: 3,
      name: `Billi Willi`,
      super: true,
      picture: `img/avatar-angelina.jpg`,
    },
    detailsMapZoom: 12,
    coordinates: [52.390955394351, 4.85309666406201],
    mapZoom: 10,
    city: `Amsterdam`,
    cityCoordinates: [52.38333, 4.9],
  },
  {
    id: 4,
    isFavorite: true,
    isPremium: true,
    features: {
      type: `house`,
      badrooms: 6,
      adults: 3,
    },
    price: 400,
    title: `TITLE - 4`,
    rating: 2.1,
    things: [
      `Wifi`,
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Own bathroom`,
      `Hair-dryer`,
      `AC`,
    ],
    previewPictures: `img/apartment-small-01.jpg`,
    pictures: [
      `img/apartment-small-03.jpg`,
      `img/apartment-small-04.jpg`,
      `img/room-small.jpg`,
      `img/studio-photos.jpg`,
    ],
    description: `The delusion emphasizes intelligent common sense, although the opposite is accepted in the official language.`,
    owner: {
      id: 4,
      name: `Agent 007`,
      super: true,
      picture: `img/avatar-angelina.jpg`,
    },
    detailsMapZoom: 12,
    coordinates: [52.3909553943515, 4.85309666406196],
    mapZoom: 10,
    city: `Amsterdam`,
    cityCoordinates: [52.38333, 4.9],
  },
];

export const reviews = [
  {
    id: 1,
    picture: `img/avatar-max.jpg`,
    name: `Spider Man`,
    rating: 2,
    review: `She carefully took a crumpled, half-torn ten-dollar bill from her purse and handed it to him. He shoved a white`,
    date: `2020-05-08T14:13:56.569Z`,
    super: true,
  },
  {
    id: 2,
    picture: `img/avatar-max.jpg`,
    name: `Aqua Man`,
    rating: 5,
    review: `Be Happy! too! She carefully took a crumpled, half-torn ten-dollar bill from her purse and handed it to him. He shoved a white`,
    date: `2020-06-08T14:30:56.569Z`,
    super: true,
  },
];

export const serverOffers = [
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: `Amsterdam`,
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
    ],
    host: {
      avatar_url: `img/1.png`,
      id: 3,
      is_pro: true,
      name: `Angelina`,
    },
    id: 1,
    images: [`img/1.png`, `img/2.png`],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8,
    },
    max_adults: 4,
    preview_image: `img/1.png`,
    price: 120,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`,
  },
  {
    bedrooms: 4,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: `Amsterdam`,
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
    ],
    host: {
      avatar_url: `img/1.png`,
      id: 3,
      is_pro: true,
      name: `Angelina`,
    },
    id: 2,
    images: [`img/1.png`, `img/2.png`],
    is_favorite: false,
    is_premium: true,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8,
    },
    max_adults: 4,
    preview_image: `img/1.png`,
    price: 100,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`,
  },
];

export const serverReviews = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatar_url: `img/1.png`,
      id: 4,
      is_pro: false,
      name: `Max`,
    },
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 2,
    rating: 4,
    user: {
      avatar_url: `img/1.png`,
      id: 4,
      is_pro: false,
      name: `Max`,
    },
  },
];

export const adaptedOffers = serverOffers.slice().map((offer) => offer);
export const adaptedReviews = serverReviews.slice().map((comment) => comment);

export const reviewData = {
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  rating: 4,
};

export const userData = {
  email: `Oliver.conner@gmail.com`,
  password: `12345678`,
};

export const authInfo = {
  avatar_url: `img/1.png`,
  email: `Oliver.conner@gmail.com`,
  id: 1,
  is_pro: false,
  name: `Oliver.conner`,
};

export const enteredInfo = {
  login: `test@mail.ru`,
  password: `qwerty12`,
};
