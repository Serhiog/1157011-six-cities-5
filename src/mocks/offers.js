export default [
  {
    city: `Moscow`,
    coordinatos: [52.3909553943508, 4.85309666406198],
    type: `Flat`,
    price: 740,
    maxCopacity: 3,
    features: [
      `wi-fi`,
      `tv`,
      `parking`,
      `Coffee machine`,
      `Dishwasher`,
      `breakfast`,
      `live music`,
    ],
    photos: [`https://picsum.photos/260/200`],
    description: `In deserunt voluptate est eiusmod cillum excepteur magna reprehenderit amet ut est.`,
    review: `Sit proident Lorem enim magna qui nostrud irure reprehenderit Lorem elit do.`,
    isPremium: false,
    isFavorite: true,
    rating: `80%`,
    rooms: 3,
    aboutOwner: `Cupidatat ad incididunt ea aute esse officia ipsum aliquip amet dolore magna.`,
    ownerPhoto: `https://picsum.photos/74/74`,
    ownerName: `Pandemia`,
    reviews: [
      {
        name: `Rudolf`,
        photo: `https://picsum.photos/54/54`,
        rating: `77%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
      {
        name: `Samuil`,
        photo: `https://picsum.photos/54/54`,
        rating: `84%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
    ],
    id: 1,
  },
  {
    city: `Cancun`,
    coordinatos: [52.369553943508, 4.85309666406198],
    type: `House`,
    price: 1380,
    maxCopacity: 7,
    features: [
      `wi-fi`,
      `tv`,
      `parking`,
      `breakfasts`,
      `elevator`,
      `security`,
      `Coffee machine`,
      `Dishwasher`,
    ],
    photos: [`https://picsum.photos/260/202`],
    description: `Velit excepteur eu cillum ullamco deserunt anim cupidatat et ex anim excepteur.`,
    review: `Aliqua est dolor nisi voluptate fugiat ipsum non exercitation. Irure sit esse duis commodo. Nostrud exercitation dolor ea voluptate officia aliqua sunt exercitation cillum est consequat velit esse. Proident ipsum ut voluptate Lorem nulla esse proident anim irure labore minim in. Ex id aute commodo incididunt reprehenderit. Qui nostrud ut sint ea occaecat qui pariatur ipsum sunt veniam eiusmod sit cillum laborum.`,
    isPremium: true,
    isFavorite: true,
    rating: `30%`,
    rooms: 6,
    aboutOwner: `Proident in labore in eiusmod anim excepteur aliquip do ipsum minim incididunt.`,
    ownerPhoto: `https://picsum.photos/74/75`,
    ownerName: `Vasya`,
    reviews: [
      {
        name: `Jack`,
        photo: `https://picsum.photos/54/53`,
        rating: `77%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
    ],
    id: 2,
  },
  {
    city: `Berlin`,
    coordinatos: [52.3909553943508, 4.929309666406198],
    type: `Hostel`,
    price: 250,
    maxCopacity: 1,
    features: [`wi-fi`, `Dishwasher`],
    photos: [`https://picsum.photos/260/201`],
    description: `Consectetur fugiat occaecat ad dolor aliqua esse.`,
    review: ` Ex id aute commodo incididunt reprehenderit. Qui nostrud ut sint ea occaecat qui pariatur ipsum sunt veniam eiusmod sit cillum laborum.`,
    isPremium: false,
    isFavorite: true,
    rating: `75%`,
    rooms: 1,
    aboutOwner: `Exercitation incididunt esse esse sit laborum pariatur fugiat voluptate consectetur sunt magna aliqua mollit nulla.`,
    ownerPhoto: `https://picsum.photos/74/76`,
    ownerName: `Petro`,
    reviews: [
      {
        name: `Elena`,
        photo: `https://picsum.photos/54/55`,
        rating: `77%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
    ],
    id: 3,
  },
  {
    city: `Paris`,
    coordinatos: [52.3809553943508, 4.939309666406198],
    type: `Bungalo`,
    price: 500,
    maxCopacity: 2,
    features: [`nature`],
    photos: [`https://picsum.photos/260/199`],
    description: `Sit velit sint cupidatat reprehenderit et irure reprehenderit officia incididunt.`,
    review: `Excepteur reprehenderit eiusmod dolor quis proident ut amet eiusmod id duis reprehenderit. Dolore labore culpa sunt do proident. Incididunt pariatur non nisi do. Do ut do est eu anim amet ipsum do dolor eiusmod nulla.Quis ullamco consectetur est pariatur fugiat ipsum id ipsum amet et eu. Exercitation aliqua irure ut velit occaecat minim. Non ad veniam cillum quis in Lorem ad.`,
    isPremium: false,
    isFavorite: false,
    rating: `90%`,
    rooms: 0,
    aboutOwner: `Fugiat pariatur sit enim excepteur do nulla.`,
    ownerPhoto: `https://picsum.photos/74/73`,
    ownerName: `Anton`,
    reviews: [
      {
        name: `Petro`,
        photo: `https://picsum.photos/54/52`,
        rating: `77%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
    ],
    id: 4,
  },
  {
    city: `Kiev`,
    coordinatos: [52.4109553943508, 4.85309666406198],
    type: `Flat`,
    price: 740,
    maxCopacity: 3,
    features: [
      `wi-fi`,
      `tv`,
      `parking`,
      `Coffee machine`,
      `Dishwasher`,
      `breakfast`,
      `live music`,
    ],
    photos: [`https://picsum.photos/260/200`],
    description: `In deserunt voluptate est eiusmod cillum excepteur magna reprehenderit amet ut est.`,
    review: `Sit proident Lorem enim magna qui nostrud irure reprehenderit Lorem elit do.`,
    isPremium: false,
    isFavorite: true,
    rating: `80%`,
    rooms: 3,
    aboutOwner: `Cupidatat ad incididunt ea aute esse officia ipsum aliquip amet dolore magna.`,
    ownerPhoto: `https://picsum.photos/74/74`,
    ownerName: `Pandemia`,
    reviews: [
      {
        name: `Rudolf`,
        photo: `https://picsum.photos/54/54`,
        rating: `77%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
      {
        name: `Samuil`,
        photo: `https://picsum.photos/54/54`,
        rating: `84%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
    ],
    id: 5,
  },
  {
    city: `Istambul`,
    coordinatos: [52.4009553943508, 4.85309666406198],
    type: `Flat`,
    price: 740,
    maxCopacity: 3,
    features: [
      `wi-fi`,
      `tv`,
      `parking`,
      `Coffee machine`,
      `Dishwasher`,
      `breakfast`,
      `live music`,
    ],
    photos: [`https://picsum.photos/260/200`],
    description: `In deserunt voluptate est eiusmod cillum excepteur magna reprehenderit amet ut est.`,
    review: `Sit proident Lorem enim magna qui nostrud irure reprehenderit Lorem elit do.`,
    isPremium: false,
    isFavorite: true,
    rating: `80%`,
    rooms: 3,
    aboutOwner: `Cupidatat ad incididunt ea aute esse officia ipsum aliquip amet dolore magna.`,
    ownerPhoto: `https://picsum.photos/74/74`,
    ownerName: `Pandemia`,
    reviews: [
      {
        name: `Rudolf`,
        photo: `https://picsum.photos/54/54`,
        rating: `77%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
      {
        name: `Samuil`,
        photo: `https://picsum.photos/54/54`,
        rating: `84%`,
        message: `Consequat eiusmod dolor anim ea anim mollit do voluptate labore deserunt est laborum excepteur culpa.`,
        date: new Date(),
      },
    ],
    id: 6,
  },
];
