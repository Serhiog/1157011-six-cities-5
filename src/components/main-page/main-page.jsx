import React from "react";
import PropTypes from "prop-types";
import OfferList from "../offers-list/offer-list";
import Map from "../map/map";
import CitiesList from "../cities-list/cities-list";
import MainEmpty from "../main-empty/main-empty";
import {connect} from "react-redux";
import {getFiltredByCityOffers} from "../../store/offers/selectors";
import {PropTypes4Offer} from "../../propConsts";
import {AuthorizationStatus} from "../../consts";

const MainPage = ({ goToFavorites, offers, isLogged, email }) => {
  const city = offers[0].city
  return (
    <div className="page page--gray page--main">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <a
                className="header__logo-link header__logo-link--active"
                href="#"
              >
                <img
                  className="header__logo"
                  src="img/logo.svg"
                  alt="6 cities logo"
                  width={81}
                  height={41}
                />
              </a>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <a
                    className="header__nav-link header__nav-link--profile"
                    onClick={goToFavorites}
                  >
                    <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                    <span className="header__user-name user__name">
                      {isLogged === AuthorizationStatus.AUTH
                        ? email
                        : `Sign In`}
                    </span>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>
      <main
        className={`${
          !offers.length ? `page__main--index-empty` : ``
        } page__main page__main--index `}
      >
        <h1 className="visually-hidden">Cities</h1>
        <CitiesList />
        {!offers.length ? (
          <MainEmpty />
        ) : (
          <div className="cities">
            <div className="cities__places-container container">
              <OfferList
                offers={offers}
                classList={`cities__places-list tabs__content`}
                classCard={`cities__place-card`}
                classImageWrapper={`cities__image-wrapper`}
              />
              <div className="cities__right-section">
                <section className="cities__map map">
                  <Map
                    offers={offers}
                    classMap={`cities__map`}
                    cityCoordinates={[
                      city.location.latitude,
                      city.location.longitude,
                    ]}
                    mapZoom={city.location.zoom}
                  />
                </section>
              </div>
            </div>
          </div>
        )}
      </main>
    </div>
  );
};

MainPage.propTypes = {
  goToFavorites: PropTypes.func,
  offers: PropTypes.arrayOf(PropTypes.shape(PropTypes4Offer)),
  isLogged: PropTypes.string,
  email: PropTypes.string.isRequired,
};

const mapToStateProps = (state) => ({
  offers: getFiltredByCityOffers(state),
  isLogged: state.user.authorizationStatus,
  email: state.user.email,
});

export {MainPage};
export default connect(mapToStateProps)(MainPage);
