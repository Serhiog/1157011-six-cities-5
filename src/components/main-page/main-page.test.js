import React from "react";
import renderer from "react-test-renderer";
import {MainPage} from "./main-page";
import {serverOffers} from "../../mocks/data";
import {BrowserRouter} from "react-router-dom";

jest.mock(`../offers-list/offer-list`, () => `OfferList`);
jest.mock(`../map/map`, () => `Map`);
jest.mock(`../cities-list/cities-list`, () => `CitiesList`);
jest.mock(`../main-empty/main-empty`, () => `MainEmpty`);

it(`MainPage with offers`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <MainPage
            goToFavorites={() => {}}
            offers={serverOffers}
            isLogged={true}
            email={`email@email.com`}
            city={{
              location: {latitude: 1, longitude: 1},
              name: `Cologne`
            }}
          />
        </BrowserRouter>
    )

    .toJSON();

  expect(tree).toMatchSnapshot();
});
