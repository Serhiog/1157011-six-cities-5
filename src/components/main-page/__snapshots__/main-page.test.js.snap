// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage render MainPage with offers 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="#"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign In
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className=" page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CitiesList />
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <OfferList
          classCard="cities__place-card"
          classImageWrapper="cities__image-wrapper"
          classList="cities__places-list tabs__content"
          offers={
            Array [
              Object {
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar_url": "img/1.png",
                  "id": 3,
                  "is_pro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "is_favorite": false,
                "is_premium": false,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "max_adults": 4,
                "preview_image": "img/1.png",
                "price": 120,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
              Object {
                "bedrooms": 4,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar_url": "img/1.png",
                  "id": 3,
                  "is_pro": true,
                  "name": "Angelina",
                },
                "id": 2,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "is_favorite": false,
                "is_premium": true,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "max_adults": 4,
                "preview_image": "img/1.png",
                "price": 100,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
            ]
          }
        />
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Map
              cityCoordinates={
                Array [
                  52.370216,
                  4.895168,
                ]
              }
              classMap="cities__map"
              mapZoom={10}
              offers={
                Array [
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatar_url": "img/1.png",
                      "id": 3,
                      "is_pro": true,
                      "name": "Angelina",
                    },
                    "id": 1,
                    "images": Array [
                      "img/1.png",
                      "img/2.png",
                    ],
                    "is_favorite": false,
                    "is_premium": false,
                    "location": Object {
                      "latitude": 52.35514938496378,
                      "longitude": 4.673877537499948,
                      "zoom": 8,
                    },
                    "max_adults": 4,
                    "preview_image": "img/1.png",
                    "price": 120,
                    "rating": 4.8,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                  Object {
                    "bedrooms": 4,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatar_url": "img/1.png",
                      "id": 3,
                      "is_pro": true,
                      "name": "Angelina",
                    },
                    "id": 2,
                    "images": Array [
                      "img/1.png",
                      "img/2.png",
                    ],
                    "is_favorite": false,
                    "is_premium": true,
                    "location": Object {
                      "latitude": 52.35514938496378,
                      "longitude": 4.673877537499948,
                      "zoom": 8,
                    },
                    "max_adults": 4,
                    "preview_image": "img/1.png",
                    "price": 100,
                    "rating": 4.8,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`MainPage with offers 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="#"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign In
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className=" page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CitiesList />
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <OfferList
          classCard="cities__place-card"
          classImageWrapper="cities__image-wrapper"
          classList="cities__places-list tabs__content"
          offers={
            Array [
              Object {
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar_url": "img/1.png",
                  "id": 3,
                  "is_pro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "is_favorite": false,
                "is_premium": false,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "max_adults": 4,
                "preview_image": "img/1.png",
                "price": 120,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
              Object {
                "bedrooms": 4,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar_url": "img/1.png",
                  "id": 3,
                  "is_pro": true,
                  "name": "Angelina",
                },
                "id": 2,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "is_favorite": false,
                "is_premium": true,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "max_adults": 4,
                "preview_image": "img/1.png",
                "price": 100,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
            ]
          }
        />
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Map
              cityCoordinates={
                Array [
                  52.370216,
                  4.895168,
                ]
              }
              classMap="cities__map"
              mapZoom={10}
              offers={
                Array [
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatar_url": "img/1.png",
                      "id": 3,
                      "is_pro": true,
                      "name": "Angelina",
                    },
                    "id": 1,
                    "images": Array [
                      "img/1.png",
                      "img/2.png",
                    ],
                    "is_favorite": false,
                    "is_premium": false,
                    "location": Object {
                      "latitude": 52.35514938496378,
                      "longitude": 4.673877537499948,
                      "zoom": 8,
                    },
                    "max_adults": 4,
                    "preview_image": "img/1.png",
                    "price": 120,
                    "rating": 4.8,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                  Object {
                    "bedrooms": 4,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatar_url": "img/1.png",
                      "id": 3,
                      "is_pro": true,
                      "name": "Angelina",
                    },
                    "id": 2,
                    "images": Array [
                      "img/1.png",
                      "img/2.png",
                    ],
                    "is_favorite": false,
                    "is_premium": true,
                    "location": Object {
                      "latitude": 52.35514938496378,
                      "longitude": 4.673877537499948,
                      "zoom": 8,
                    },
                    "max_adults": 4,
                    "preview_image": "img/1.png",
                    "price": 100,
                    "rating": 4.8,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
